packages/background/src/controllers/AccountTrackerController.ts

```diff
@@ -179,17 +179,28 @@ export class AccountTrackerController extends BaseController<AccountTrackerState
 
         this._tokenController.on(
             TokenControllerEvents.USER_TOKEN_CHANGE,
-            async (accountAddress: string, chainId: number) => {
+            async (
+                accountAddress: string,
+                chainId: number,
+                tokenAddress?: string
+            ) => {
                 try {
-                    // Update the account balances
-                    await this.updateAccounts({
-                        addresses: [accountAddress],
-                        assetAddresses:
-                            await this._tokenController.getUserTokenContractAddresses(
-                                accountAddress,
-                                chainId
-                            ),
-                    });
+                    // If there is a tokenAddress, we update that asset balance, else we update account balances
```

Comment:
This comment is on the other way around of the implementation. 

I would do what the comment says, If there is a tokenAddress then ... It seems more clear